automation_tools/satellite6/capsule.py

```diff
@@ -0,0 +1,398 @@
+                config.admin_password,
+                config.organization_label,
+                config.content_view,
+            )
+        )
+    else:
+        raise ImproperlyConfigured(
+            'An activation key or content_view name is required.')
+    run('yum repolist')
+
+
+@task
+def capsule_installer(
+        foreman_oauth_key, foreman_oauth_secret, pulp_oauth_secret):
+    """Install and run capsule-installer."""
+    config = _get_config()
+    run('yum -y install capsule-installer')
+    run(
+        'capsule-installer -v --certs-tar *-certs.tar --parent-fqdn {} '
+        '--pulp true --pulp-oauth-secret {} --puppet true --puppetca true '
+        '--foreman-oauth-secret {} --foreman-oauth-key {} '
+        '--register-in-foreman true --qpid-router true --reverse-proxy true'
+        .format(
+            config.server.hostname,
+            pulp_oauth_secret,
+            foreman_oauth_secret,
+            foreman_oauth_key
+        )
+    )
+
+
+@task
+def hammer(command):
+    """Run hammer -u <admin_user> -p <admin_password> --output json <command>.
+
+    :param str command: The hammer subcommand to run.
+    :return: Return a JSON decoded object containing the result of the command.
+        The returned object will exhibit ``failed`` and ``succeeded`` boolean
+        attributes specifying whether the command failed or succeeded, and will
+        also include the return code as the ``return_code`` attribute.
+
+    """
+    config = _get_config()
+    command_result = run(
+        'hammer --username {} --password {} --output json {}'
+        .format(config.admin_user, config.admin_password, command),
+        quiet=True
+    )
+    result = _lower_dict_keys(json.loads(command_result))
+    print(result)
+    if isinstance(result, list):
+        result = _AttributeList(result)
+    elif isinstance(result, dict):
+        result = _AttributeDict(result)
+    result.succeeded = command_result.succeeded
+    result.failed = command_result.failed
+    result.return_code = command_result.return_code
+    return result
+
+
+@task
+def hammer_capsule_lcenvs(capsule_id):
+    """Get the available lifecycle environments of a capsule.
+
+    :param capsule_id: The capsule ID to get the availables lifecycle
+        environments.
+    :returns: A list of lifecycle environment dictonaries. For example:
+        ``[{u'organization': u'Default Organization', u'id': 1,
+        u'name': u'Library'}]``.
+    :rtype: list
+
+    """
+    return hammer(
+        'capsule content available-lifecycle-environments --id {}'
+        .format(capsule_id),
+    )
+
+
+@task
+def hammer_capsule_add_lcenv(capsule_id, lcenv_id):
+    """Add the lifecycle environment to the capsule.
+
+    :param capsule_id: The capsule ID to add the lifecycle environment.
+    :param lcenv_id: The lifecycle environment ID to add to the capsule.
+
+    """
+    return hammer(
+        'capsule content add-lifecycle-environment --environment-id {} --id {}'
```

Comment:
That is just to support python 2.6, I don't see a real benefit of supporting that version. The numbering is required only if that makes sense, otherwise it will use the order of the arguments passed to format, so less typing.